README.rst

```diff
@@ -1,81 +1,110 @@
+              ...
+  [snip]
+  $ west flash -h
+  usage: west flash [-h] [-H] [-d BUILD_DIR] ...
+  [snip]
 
-  $ west init your/zephyr/install-dir
-  $ cd your/zephyr/install-dir
-  $ west <command>
+Test Suite
+----------
 
-The ``west init`` call will:
+To run the test suite, run this from the west repository::
 
-- create your/zephyr/install-dir
-- clone the Zephyr manifest repository (whose URL can be overridden
-  with the ``-u`` option, and branch with ``--mr`` / ``--manifest-rev``)
-- clone the latest West repository (the URL override is ``-w``, and
-  revision/branch override is ``--wr`` / ``--west-rev``)
+  pip3 install -r tests_requirements.txt
+  python3 setup.py test
 
-Running ``west <command>`` from :file:`your/zephyr/install-dir` or
-underneath it will invoke west in "wrapper" mode: any commands other
-than ``init`` will be delegated to the West tree pulled by ``west
-init``.
+Hacking on West
+---------------
 
-This arrangement may seem familiar to Android (platform, not app)
-developers. The source management features of West were indeed
-inspired by the Android Repo tool's features, but West makes
-significant departures from Repo's behavior.
+West is distributed as two Python packages:
 
-Alternative Usage
------------------
+1. A ``bootstrap`` package, which is distributed via PyPI. Running
+   ``pip3 install west`` installs this **bootstrapper package only**.
+2. The "main" ``west`` package, which is fetched by the bootstrapper
+   when ``west init`` is run.
 
-If you don't want to change your system outside of cloning this
-repository, you can also clone West's Git repository and run the
-package as a module:
+This somewhat unusual arrangement is because:
 
-$ python3 -m west
+- One of west's jobs is to manage a Zephyr installation's Git
+  repositories, including its own.
+- It allows easy customization of the version of west that's shipped
+  with non-upstream distributions of Zephyr.
+- West is experimental and is not stable. Users need to stay in sync
+  with upstream, and this allows west to automatically update itself.
 
-Only do this if you know what you're doing (and make sure to install
-the requirements as specified by the install_requires line in
-setup.py).
+To initialize west from a non-default location::
 
-Test Suite
-----------
+  west init -w https://example.com/your-west-repository.git
+
+You can also add ``--west-rev some-branch`` to use ``some-branch``
+instead of ``master``.
+
+To use another manifest repository (optionally with ``--mr
+some-manifest-branch``)::
+
+  west init -u https://example.com/your-manifest-repository.git
+
+After ``init`` time, you can hack on the west tree in ``zephyrproject``.
+
+To install everything in development mode instead, clone the west
+repository and run this from the top level directory::
+
+  pip3 install -e .
+
+(On Linux, make sure ``~/.local/bin`` is in your ``PATH`` if you go
+this route.)
```

Comment:
Maybe this could be added at the end, within the parenthesis: "Alternatively, you can run ``west`` with ``python -m west``."